lib/datasets/job.rb

```diff
@@ -29,5 +29,17 @@ def self.deserialize(*args)
       raise NotImplementedError
     end
 
+    private
+
+    # Reinvent HashWithIndifferentAccess
+    def self.deserialize_volume(hash)
+      Volume.new(
+        namespace: hash[:namespace] || hash["namespace"],
+        id: hash[:id] || hash["id"],
+        access_profile: hash[:access_profile] || hash["access_profile"],
+        right: hash[:right] || hash["right"]
```

Comment:
For these four parameters, this seems fine.  It's straight forward enough.  The alternative map and reduce method would have been fine too.  If we end up doing this in two places, it might be worth our while to then extract that to a function for keys_to_symbols.